### PR TITLE
Integrate Document configurability

### LINKED ISSUE #748: Form configuration android changes
See related documentation on [Jira](https://smartregister.atlassian.net/browse/RVL-957)

### PR DESCRIPTION
**Pull request by Rkareko**
Resolves #748

### CODE REVIEW

File: opensrp-reveal/src/main/java/org/smartregister/reveal/util/RevealJsonFormUtils.java [hunk 1]
```diff
@@ -145,14 +147,26 @@ public JSONObject getFormJSON(Context context, String formName, BaseTaskDetails
...
+                String structType = structureType;
+                if (StringUtils.isBlank(structureType)) {
+                    structType = Constants.StructureType.NON_RESIDENTIAL;
+                }
+                formString = formString.replace(JsonForm.STRUCTURE_PROPERTIES_TYPE, structType);
             }
-            formString = formString.replace(JsonForm.STRUCTURE_PROPERTIES_TYPE, structType);
+
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (dubdabasoduba) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/util/RevealJsonFormUtils.java:169]: Use timber

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

File: opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java [hunk 2]
```diff
@@ -81,6 +83,8 @@ public void onCreate(SQLiteDatabase database) {
         TaskRepository.createTable(database);
         LocationRepository.createTable(database);
         StructureRepository.createTable(database);
+        ClientFormRepository.createTable(database);
```
Comment (ekigamba) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java:86]: Add migration scripts for these two DBs if you are going to release to running prod

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadow.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

File: opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadow.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.smartregister.sync.CloudantDataHandler;
+
+/**
+ * Created by Richard Kareko on 5/29/20.
+ */
+
+@Implements(CloudantDataHandler.class)
+public class CloudantDataHandlerShadow {
```
Comment (keymane) [inline opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadow.java:15]: Codacy Issue found: [The utility class name 'CloudantDataHandlerShadow' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929)

File: opensrp-reveal/build.gradle [hunk 4]
```diff
@@ -188,17 +188,7 @@ dependencies {
...
-        exclude group: 'com.android.support', module: 'appcompat-v7'
-        exclude group: 'com.android.support', module: 'cardview-v7'
-        exclude group: 'com.android.support', module: 'support-media-compat'
-        exclude group: 'com.android.support', module: 'support-v4'
-        exclude group: 'com.android.support', module: 'design'
-        exclude group: 'io.ona.rdt-capture', module: 'lib'
-    }
-
-    implementation('org.smartregister:opensrp-client-family:1.2.32-SNAPSHOT@aar') {
+    implementation('org.smartregister:opensrp-client-family:1.2.34-SNAPSHOT@aar') {
```
Comment (dubdabasoduba) [inline opensrp-reveal/build.gradle:191]: Don't you need native forms again or you using the version that comes with core?

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadow.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

File: opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadow.java (hunk 3 above)
Comment (keymane) [inline opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadow.java:15]: Codacy Issue found: [The utility class name 'CloudantDataHandlerShadow' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadowUtils.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

File: opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java (hunk 2 above)
Comment (Rkareko) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java]: done

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/test/java/org/smartregister/reveal/shadow/CloudantDataHandlerShadowUtils.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/job/RevealJobCreator.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5569929&bid=18265038)

**Pull request merged**